### PR TITLE
Make audit log optional

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Makes the audit logging optional as it creates a very high logvolume.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

/assign @zreigz @xrstf 
CC @eqrx

### CODE REVIEW

File: api/pkg/crd/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -90,6 +90,9 @@ type ClusterSpec struct {
 	Openshift *Openshift `json:"openshift,omitempty"`
 
 	UsePodSecurityPolicyAdmissionPlugin bool `json:"usePodSecurityPolicyAdmissionPlugin,omitempty"`
+
+	// AuditLogging holds auditlogging settings
```
Comment (xrstf) [inline api/pkg/crd/kubermatic/v1/cluster.go]: This comment is useless. :-)

File: api/pkg/crd/kubermatic/v1/cluster.go [hunk 2]
```diff
@@ -90,6 +90,9 @@ type ClusterSpec struct {
...
+	AuditLogging *AuditLoggingSettings `json:"auditlogging,omitempty"`
```
Comment (xrstf) [inline api/pkg/crd/kubermatic/v1/cluster.go]: Can you make this camelCase (`auditLogging`), so it's consistent with the other fields?

**Comment by xrstf** [comment]
/approve

Will /lgtm once we resolved my two points.

**Comment by alvaroaleman** [comment]
@xrstf Updated
@zreigz Any idea whats up with the swagger verification? Latest master is merged in this PR

**Comment by alvaroaleman** [comment]
/retest
@zreigz Any chance you can have a look?

**Comment by zreigz** [comment]
@alvaroaleman yes, I will take a look, busy day a lot of meetings.

**Comment by zreigz** [comment]
sorry for delay
/approve

**Pull request merged**